src/Mutator/MutatorGenerator.php

```diff
@@ -0,0 +1,137 @@
+<?php
+/**
+ * Copyright © 2017-2018 Maks Rafalko
+ *
+ * License: https://opensource.org/licenses/BSD-3-Clause New BSD License
+ */
+
+declare(strict_types=1);
+
+namespace Infection\Mutator;
+
+use Infection\Config\Exception\InvalidConfigException;
+
+class MutatorGenerator
+{
+    /**
+     * @var array
+     */
+    private $mutatorSettings;
+
+    /**
+     * @var array
+     */
+    private $mutatorList = [];
+
+    public function __construct(array $mutatorSettings = [])
+    {
+        $this->mutatorSettings = $mutatorSettings;
+    }
+
+    /**
+     * Will create an array of mutators, keyed by their name
+     *
+     * @throws InvalidConfigException
+     *
+     * @return Mutator[]
+     */
+    public function create(): array
+    {
+        if (empty($this->mutatorSettings)) {
+            $this->mutatorSettings = [
+                '@default' => true,
+            ];
+        }
+        $this->mutatorList = [];
+
+        foreach ($this->mutatorSettings as $mutator => $setting) {
```

Comment:
in this loop variable `$mutator` can be either mutator or profile. Does it make sense to rename it to something like `$mutatorOrProfile` for better readability?